libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFAction.m

```diff
@@ -541,8 +541,8 @@ - (id)contentFromData:(NSData*)data fromResponse:(NSHTTPURLResponse*)response er
         content = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:&jsonParseError];
     }
     
-    // If it's an error here, it's a basic parsing error.
-    if (jsonParseError && error) {
+    // If it's an error here, it's a basic parsing error and error is not 500 internal server error.
+    if (jsonParseError && error && response.statusCode != 500) {
```

Comment:
This seems...an odd approach.  Wouldn't it make more sense to put the status code check in `completeOperationWithResponse:`, before reading the content?  Because I'm reading this as: 500 errors should be treated independently of response content parsing.  If that's the case, this doesn't belong here.